comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -7,54 +7,139 @@
+
+        // Return the nesting depth of the given directory. A directory containing only files (no subdirectories) has a depth of 0.
+        public static int GetDepth(string directory)
+        {
+            int depth = 0;
+            string[] array_of_files = Directory.GetDirectories(directory);
+            bool loop = true;
+            for (int s = 0; array_of_files.Length > s; s++)
+            {
+                if (array_of_files.Length >= 1)
+                {
+                    depth += 1;
+                    array_of_files = Directory.GetDirectories(array_of_files[s]);
+                }
+            }
+            return depth;
+        }
+
+        // Get the path and size (in bytes) of the smallest file below the given directory
+        public static Tuple<string, long> GetSmallestFile(string directory)
+        {
+            long Smallest_byte = 9223372036854775807;
+            string Location_of_file = "";
+            string [] file = Directory.GetFileSystemEntries(directory);
+            for (int s = 0; file.Length > s; s++)
+            {
+                if (IsDirectory(file[s]) == true)
+                {
+                    Location_of_file && Smallest_byte = GetSmallestFile(file[s]);
+                }
+                else if (GetFileSize(file[s]) < Smallest_byte)
+                {
+                    Smallest_byte = GetFileSize(file[s]);
+                    Location_of_file = file[s];
+
+                }
+            }
+            return Smallest_byte;
+        }
+    
+
+    // Get the path and size (in bytes) of the largest file below the given directory
+    public static Tuple<string, long> GetLargestFile(string directory)
+    {
+            long Largest_byte = 0;
+            string Location_of_file = "";
+            string [] file = Directory.GetFileSystemEntries(directory);
+            for (int s = 0; file.Length > s; s++)
+            {
+                if (IsDirectory(file[s]) == true)
+                {
+                    Location_of_file && Largest_byte = GetLargestFile(file[s]);
+                }
+                else if (GetFileSize(file[s]) < Largest_byte)
+                {
+                    Largest_byte = GetFileSize(file[s]);
+                    Location_of_file = file[s];
+
+                }
+            }
+            return Largest_byte;
+        }
+
+    // Get all files whose size is equal to the given value (in bytes) below the given directory
+    public static IEnumerable<string> GetFilesOfSize(string directory, long size)
+    {
+            string[] same_file_size;
+            string[] same_file_size_plus;
+            string[] file = Directory.GetFileSystemEntries(directory);     
+            for (int s = 0; file.Length > s; s++)     
+            {
+                if (IsDirectory(file[s]) == true)     
+                {
+                    same_file_size_plus = GetFileSystem(file[s]);     
+                    same_file_size.Add(same_file_size_plus);     
```

Comment:
You can't `Add` to an array, because arrays have a fixed size. You'd need to make `same_file_size` a `List<string>` to be able to add elements to it